README.md

```diff
@@ -1,8 +1,195 @@
+The *addDeviceType()* method creates a device type within your organization.  The device type is needed for all device interractions, for example creating a device, or posting device data.  This method takes one required parameter: a table with the info to create a device type, and two optional parameters: a *headers* table and a *cb* function.
+
+#### typeInfo Table
+| key | value data type | Required | description |
+| ----- | ------------ | ----------- | --------------- |
+| *id* | string | Yes | The device type id is used to uniquely identify the device type (36 character limit) |
+| *description* | string | No | The device type description can be used for a more descriptive way of identifying the device type |
+| *classId* | string | No | Accepted values are "Device" or "Gateway", if you do not pass in this value defaults to "Device" |
+| *deviceInfo* | table | No | Keys in the deviceInfo table define attributes to be used as a template for new devices that are assigned this device type. Attributes you do not define may still be edited individually on devices that are assigned this device type. |
+| *metadata* | table | No | Meta data you wish to have associated with the device type |
+
+##### Example Code:
+```squirrel
+typeInfo <- {"id": "EnvTail", "description" : "Electric Imp Environmental Sensor Tail"};
+watson.addDeviceType(typeInfo, function(error, response) {
+    if (error) server.error(error);
+})
+```
+
+### getDeviceType(*typeID[, headers][, cb]*)
+The *getDeviceType()* method requests the details for the device type specified.  This method takes one required parameter: the device's *typeID*, and two optional parameters: a *headers* table and a *cb* function.
+
+##### Example Code:
+```squirrel
+local typeID = "EnvTail";
+local typeInfo = {"id": typeID, "description" : "Electric Imp Environmental Sensor Tail"};
+
+watson.getDeviceType(typeID, function(error, response) {
+    // if device type doesn't exist create it
+    if (error == watson.MISSING_RESOURCE_ERROR) {
+        watson.addDeviceType(typeInfo);
+    }
+})
+```
+
+### addDevice(*typeID, device[, headers][, cb]*)
+The *addDevice()* method creates a device of the specified device type.  This method takes two required parameters: the device's *typeID* and a table containing information about the *device*, and two optional parameters: a *headers* table and a *cb* function.
+
+#### Device Table
+| key | value data type | Required | description |
+| ----- | ------------ | ----------- | --------------- |
+| *deviceId* | string | Yes | The device id is used to uniquely identify the device. |
+| *authToken* | string | No | Authentication token, will be generated if not supplied. |
+| *deviceInfo* | table | No | See Device Info Table below for more details.  All keys in the deviceInfo table are optional.  |
+| *location* | table | No | See Location Table below for more details. |
+| *metadata* | table | No | Meta data you wish to have associated with the device type. |
```

Comment:
Watson documentation doesn't go into detail, so I didn't either.  Not sure what a good example is.